lib/internal/streams/duplexify.js

```diff
@@ -102,12 +106,11 @@ module.exports = function duplexify(body, name) {
 
     const then = value?.then;
     if (typeof then === 'function') {
-      let d;
-
       const promise = FunctionPrototypeCall(
         then,
         value,
         (val) => {
+          destroyer(d, null);
```

Comment:
```suggestion
          destroyer(d);
```

If `err` is `null`, it's treated the same as if it's undefined:
https://github.com/nodejs/node/blob/3c5ceff85fb90b265a85270bba38085f45c7f25e/lib/internal/streams/destroy.js#L327